// @codux-boards/react-blueprint/boards/Toast/danger.board.tsx
import { createBoard } from '@wixc3/react-board';
import { Button, OverlayToaster, Toaster } from '@blueprintjs/core';

// board cover
import cover from '../../../../assets/toast/danger.svg';

const toaster: Toaster = OverlayToaster.create({
    autoFocus: true,
    canEscapeKeyClear: true,
    maxToasts: 3,
    position: 'top',
    usePortal: true,
});

export default createBoard({
    name: 'Danger',
    Board: () => {
        return (
            <Button
                intent="danger"
                onClick={() =>
                    toaster.show({
                        icon: 'warning-sign',
                        intent: 'danger',
                        message:
                            'You do not have permissions to perform this action. \
                            Please contact your system administrator to request the appropriate access rights.',
                    })
                }
            >
                Toast
            </Button>
        );
    },
    tags: ['danger', 'toast'],
    cover,
});
